agent/cloudinit/file_writer_test.go

```diff
@@ -97,4 +97,38 @@ var _ = Describe("FileWriter", func() {
 		Expect(string(buffer)).To(Equal(fileOriginContent + file.Content))
 
 	})
+
+	It("should return error with invalid owner format", func() {
+		file := cloudinit.Files{
+			Path:        path.Join(workDir, "file1.txt"),
+			Encoding:    "",
+			Owner:       "root",
+			Permissions: "",
+			Content:     "some-content",
+			Append:      false,
+		}
+
+		err := cloudinit.FileWriter{}.MkdirIfNotExists(workDir)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = cloudinit.FileWriter{}.WriteToFile(&file)
+		Expect(err).To(MatchError("invalid owner format 'root'"))
+	})
+
+	It("should return error with unknown owner", func() {
```

Comment:
There is use of chown to change the owner in that function. I was getting the error `chown /tmp/file_writer_ut2246122839/file1.txt: operation not permitted` as the test are not run with permissions, still not sure how we can test this. Also this PR is to deprecate the installer flow, these tests were added just to avoid decrease in current coverage. I can take up tests in another PR 🙂

Reply:
> I can take up tests in another PR


Make sense 👍